File: simplified-books-controller-api/src/main/java/org/nypl/simplified/books/controller/api/BookUnsupportedTypeException.java
package org.nypl.simplified.books.controller.api;

import com.io7m.jnull.NullCheck;

/**
 * An exception indicating that a book cannot be fulfilled because of an
 * unsupported type.
 */

public final class BookUnsupportedTypeException extends BookException
{
  private final String type;

  /**
   * Construct an exception.
   *
   * @param in_type The unsupported file type
   */

  public BookUnsupportedTypeException(final String in_type)
  {
    super(in_type);
    this.type = NullCheck.notNull(in_type);
  }

  /**
   * @return The type of unsupported book
   */

  public String getType()
  {
    return this.type;
  }
}

File: simplified-books-controller-api/src/main/java/org/nypl/simplified/books/controller/api/BookUnsupportedTypeException.kt
package org.nypl.simplified.books.controller.api

import one.irradia.mime.api.MIMEType

/**
 * An exception indicating that a book cannot be fulfilled because of an
 * unsupported type.
 */

class BookUnsupportedTypeException(
  val type: MIMEType
) : BookException(type.fullType)
